wpsc-includes/variations.class.php

```diff
@@ -296,3 +296,39 @@ function wpsc_get_child_object_in_terms_var( $parent_id, $terms, $taxonomies, $a
 		return false;
 	}
 }
+
+/**
+ * Given array of variation selections this works through the terms and returns the product_id for the matching variation
+ *
+ * @since 4.0
+ *
+ * @param array     $variations     required            The array of variation selections
+ * @uses wpsc_get_child_objects_in_terms()              Given $product_id and product params this returns the variation product id
+ * @return int      $product_id                         The ID of the variation product
+ */
+function wpsc_get_product_id_from_variations( $variations ) {
+
+	$provided_parameters = array();
+
+	foreach ( (array) $variations['variation'] as $key => $variation ) {
+		$provided_parameters['variation_values'][ (int) $key ] = (int) $variation;
+	}
+
+	if ( count( $provided_parameters['variation_values'] ) > 0 ) {
+		$variation_product_id = wpsc_get_child_object_in_terms( $product_id, $provided_parameters['variation_values'], 'wpsc-variation' );
```

Comment:
I changed the variable name here but not in the main wpsc_add_to_cart function at this point. In my mind changing it there would be it's own PR because it has a bunch of changes to me made that could break things and it wouldn't affect whether this works or not.